### ISSUE TITLE
Serving binary file causes error

### ISSUE THREAD

**Issue by mmcdermo** [awslabs/chalice#250]
When serving a static mp3, I'm causing an error in chalice. Not sure if the problem is with my configuration, or if this is even an intended use case for Chalice at the moment. 

```python
from chalice import Chalice, Response
import boto3

boto3.set_stream_logger('')
app = Chalice(app_name='sick-beats-music.xyz') if True ^ False else boto3.client('lambda')
app.debug = True ^ False

if app is None or app == None:
    raise Exception("App should not be none")

bucket_name = "sick-beats"
file_name = "laying-it-down.mp3"

@app.route('/')
def index():
    s3 = boto3.resource('s3')
    object = s3.Object(bucket_name,file_name)
    body = object.get()["Body"].read()
    return Response(body=body,
                    status_code=200,
                    headers={'Content-Type': 'audio/mpeg',
                             'Content-length': str(len(body)),
                             'Content-Disposition': 'attachment; filename="%s"' % file_name,
                             'Content-Transfer-Encoding': 'binary'
                    })
```

Here's the error I'm getting:
```
2017-03-08 22:15:13.384000 b15715 An error occurred during JSON serialization of response: 'utf8' codec can't decode byte 0x82 in position 144: invalid start byte
Traceback (most recent call last):
  File "/usr/lib64/python2.7/json/__init__.py", line 251, in dumps
    sort_keys=sort_keys, **kw).encode(obj)
  File "/usr/lib64/python2.7/json/encoder.py", line 207, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib64/python2.7/json/encoder.py", line 270, in iterencode
    return _iterencode(o, 0)
UnicodeDecodeError: 'utf8' codec can't decode byte 0x82 in position 144: invalid start byte
```

UPDATE:
At the suggestion of an esteemed colleague, I'm now passing the length of the file contents as a string rather than an int, which was causing the original error. Now I'm getting the above utf8 decode error. Can chalice currently serve content types other than utf-8 encoded text?

**Comment by ogrodnek** [comment]
Came across this while checking the status of binary data support :).

Issue #223 (Support binary data) is still open.

One other thing to keep in mind is that api gateway has a 10MB payload size -- http://docs.aws.amazon.com/apigateway/latest/developerguide/limits.html -- not sure if the mp3s you are serving will be smaller than this?

Not sure if you've ruled this out already, but one potential workaround is you could redirect to the file in s3 (if it's publicly readable), or maybe even generate an s3 signed URL to redirect to...

**Comment by haginaga** [comment]
I post new PR #286 .
I hope it will be of your help.

**Comment by stealthycoin** [comment]
Fixed with #352
